README.md

```diff
@@ -34,9 +47,69 @@ For example, to show the working tree status of the repository:
 
     $ ambit git status
 
+### Environment variables
+
+Optionally, 3 environment variables can be used to set custom paths.
+If a variable is not set, it will take up its default value as outlined:
+
+| Environment Variable | Description                            | Default                              |
+| -------------------- | -------------------------------------- | ------------------------------------ |
+| AMBIT_HOME_PATH      | Starting path of symlink destinations. | Home directory                       |
+| AMBIT_CONFIG_PATH    | Path to configuration file.            | `${HOME}/.config/ambit/config.ambit` |
+| AMBIT_REPO_PATH      | Path to dotfile repository directory.  | `${HOME}/.config/ambit/repo`         |
+
 ## Configuration
 
-`ambit` will search for a configuration file located at `${HOME}/.config/ambit/config.ambit`.
+The purpose of the configuration file is to set the paths of the symlinks.
+A symlink is defined with two parts: an existing file relative to `AMBIT_REPO_PATH`, and its destination relative to the system's home directory.
+
+### Configuration examples
+
+#### Basic match
+
+Symlink `${HOME}/host.txt -> ${AMBIT_REPO_PATH}/a/repo.txt`:
+
+    a/repo.txt => host.txt;
+
+#### Implicit match
+
+If no `=>` operator is provided, it is assumed that the path given is both the `HOME` and `REPO` path.
+
+Symlink `${HOME}/.config/ambit/config.ambit -> ${AMBIT_REPO_PATH}/.config/ambit/config.ambit`:
+
+    .config/ambit/config.ambit;
+
+#### Variant Expression
+
+Symlink:
+
+*   `${HOME}/.config/bat/bat.conf -> ${AMBIT_REPO_PATH}/.config/bat/bat.conf`
+*   `${HOME}/.config/nvim/init.vim -> ${AMBIT_REPO_PATH}/.config/nvim/init.vim`
+
+<!---->
+
+    .config/[
+        bat/bat.conf,
+        nvim/init.vim
+    ];
+
+#### Match expressions
+
+Conditionally symlink by os and host:
+
+    {os(linux): .Xresources};
+
+    {host(plamorg): .zshrc};
```

Comment:
You may want to have an example with only one layer of `{ }` and with multiple choices.